tools/topology/topology1/sof-tgl-max98357a-rt5682.m4

```diff
@@ -293,7 +297,10 @@ PCM_PLAYBACK_ADD(HDMI1, 2, PIPELINE_PCM_5)
 PCM_PLAYBACK_ADD(HDMI2, 3, PIPELINE_PCM_6)
 PCM_PLAYBACK_ADD(HDMI3, 4, PIPELINE_PCM_7)
 PCM_PLAYBACK_ADD(HDMI4, 5, PIPELINE_PCM_8)
+# No echo reference for 2-way speakers
```

Comment:
Is this really acceptable from a product perspective? I thought all Chrome platforms provided an echo reference? @cujomalainey can you comment?

Reply:
Nope, we don't use an echo reference unless there is a reason.

That being said, if this is a 2 way, do we have a speaker protection algorithm that needs a echo ref?

Reply:
Usually speaker protection only needs I/V feedback, the echo reference can use the same path in different slots but that's a different usage/purpose. I/V feedback is to avoid blowing the speakers, echo reference to deal with acoustic coupling between speaker and mic/. Different problems.

I am still surprised that a regular 1-way speaker based solution will use an echo reference and a 2-way solution will not. One would think that adding more capabilities would not come at the expense of existing features?

Reply:
Yes, just curious if this was like the 98373 where we had to pull the stream from userspace so algorithm would get data.

Regardless if there audio is being changed I think we should probably provide an echo ref for the AEC

Reply:
@cujomalainey well that's what I was trying to say earlier but you mentioned that 'we don't use an echo reference unless there is a reason'.

AEC is a pretty good reason :-)

Reply:
@plbossart ah gotcha, our AEC isn't fully setup yet, but when it does come, it is will be needed :)